src/wallet/coinselection.cpp

```diff
@@ -48,20 +49,33 @@ bool SelectCoinsBnB(std::vector<CInputCoin>& utxo_pool, const CAmount& target_va
         remaining += utxo.txout.nValue;
     }
 
+    // Best solution
+    CAmount curr_waste = 0;
+    std::vector<std::pair<bool, bool>> best_selection;
+    CAmount best_waste = MAX_MONEY;
+
     // Depth first search to find
     while (!done)
     {
         if (tries <= 0) { // Too many tries, exit
-            return false;
+            break;
         } else if (value_ret > target_value + cost_of_change) { // Selected value is out of range, go back and try other branch
             backtrack = true;
+        } else if (curr_waste > best_waste) { // Don't select things which we know will be more wasteful
```

Comment:
I believe this must be:

`curr_waste > best_waste && (fee_vec[depth] - long_term_fee_vec[depth]) > 0`

Otherwise, you'd be prematurely stopping when a transaction is created with fees below the long-term fee estimate.

Reply:
Maybe I misunderstand what `long_term_fee` is, but how can it be lower than long term fee estimate? I thought that long term fee looks for confirmation target 1008, thus all smart fee estimates will be only bigger than the long term estimate...

Reply:
@Xekyo I agree with @runn1ng. I don't see how the long term fee can be less than the current fee.